application/models/TemplateConfiguration.php

```diff
@@ -472,12 +472,13 @@ public function search()
     }
 
     /**
-     * @todo document me
+     * Retrieves a list of models based on the current search/filter conditions.
      *
+     * @param integer $gsid Survey group, else get global
      * @return CActiveDataProvider
      * @throws Exception
      */
-    public function searchGrid()
+    public function searchGrid($gsid = null)
```

Comment:
Perhaps will be better to add here typehint instead param annotation?

Reply:
integer or null ? 
https://www.php.net/manual/en/language.types.declarations.php#language.types.declarations.nullable

Seems OK (PHP7.1)

Reply:
Added , BUT : here : the value are set by dev: if it's not a integer or null : think we must send a 500 error in all condition : 500 error : there are a development broken here ;)